### DISCUSSION TITLE
nuget: Support prerelease packages with floating versions

### DISCUSSION THREAD

**Discussion by Tobias-08** [renovatebot/renovate#11074] [Request Help]
Hi,

according to the [nuget spec](https://github.com/NuGet/Home/wiki/Support-pre-release-packages-with-floating-versions#common-scenarios) nuget supports scenarios like this:

- PackageReference with version="1.2.0-*"
- results in 1.2.0 being preferred if this stable version exists
- results in (e. g.) 1.2.0-rc.1 being preferred if no stable version exists

Is it possible to configure renovate to support this scenario? I want renovate to only look at the stable part und update to e. g. 1.2.1-* (leaving the -* untouched)?

Thanks

**Comment by Tobias-08** [comment]
Reading the [docs](https://docs.renovatebot.com/configuration-options/#rangestrategy) once again I would expect `"rangeStrategy": "replace"` to do what I intended. But it does not seem to have an effect for nuget: `-*` is still replaced by renovate.

Any ideas or is this a known functional gap/missing feature in the nuget implementation?

**Reply by viceice** [reply]
You can disable nuget manager for those deps and use a regex manager as workaround. Or provide a pr to extend renovate nuget versioning. 🙃

**Reply by rarkins** [reply]
I think we'd need to recognize this properly as a range and add tests cases like the above. Feature request issue + reproduction repo as next step

**Reply by Tobias-08** [reply]
OK. Should I raise a feature request or what is the preferred process?

**Reply by rarkins** [reply]
Yes, please create a minimal reproduction repo, then an issue using the feature request template and link to the reproduction

**Reply by Tobias-08** [reply]
Opened feature request #11127
